src/main/java/eu/clarin/cmdi/rasa/linkResources/impl/ACDHCheckedLinkResource.java

```diff
@@ -138,77 +140,93 @@ private PreparedStatement getPreparedStatement(String defaultQuery, Optional<Che
             throw new IllegalArgumentException("start and end can't less than or equal to 0 at the same time.");
         }
 
-        CheckedLinkFilter filter;
-        if (filterOptional.isPresent()) {
-            filter = filterOptional.get();
-            filter.setStart(start);
-            filter.setEnd(end);
-        } else {
-            filter = new ACDHCheckedLinkFilter(start, end);
-        }
+        final Optional<CheckedLinkFilter> filter
+                = filterOptional
+                        //filter was provided, combine with other params
+                        .map(f -> f.setStart(start).setEnd(end)) //TG: do we really want to modify the passed filter??? we could also clone
+                        //no filter was provided, create default filter 
+                        .or(() -> Optional.of(new ACDHCheckedLinkFilter(start, end)));
+
+        return get(filter);
 
-        return get(Optional.of(filter));
     }
 
     @Override
     public Map<String, CheckedLink> get(Collection<String> urls, Optional<CheckedLinkFilter> filter) throws SQLException {
+        if (urls.isEmpty()) {
+            return Collections.emptyMap();
+        } else {
+            //if urlCollection is given, this is how all these from the collection are returned:
+            //example query: select * from status where url in ('www.google.com','www.facebook.com');
+            //construct a param list for URLs
+            final StringJoiner queryInClauseJoiner = new StringJoiner(",", " url IN (", ")");
+            //add a '?' for each URL
+            urls.forEach((url) -> queryInClauseJoiner.add("?"));
+            final String queryInClause = queryInClauseJoiner.toString();
+
+            final String defaultQuery = "SELECT * FROM status WHERE" + queryInClause;
+
+            //callback to add actual URLs as parameters to prepared statement
```

Comment:
@coy123 I hope you don't find this approach too ugly; I'm not super happy and I still think this might be a smell but either way I think it's really better to avoid having the current/previous approach of just constructing the query with parameters in place in the StringBuilder; URLs can have all kinds of weird characters inside that mess up the query and depending on how this library is used there is the risk of SQL injection. We can also try to radically rethink the design or perhaps see if a library could help us solve this in a more elegant way.